### ISSUE TITLE
Error "conflicting dependencies domain_name" when trying to install plugin

### ISSUE THREAD

**Issue by vzabawski** [hashicorp/vagrant#8948]
### Vagrant version
$ vagrant -v
Vagrant 1.9.8

### Host operating system
Windows 10 x64 Build 1703

### Guest operating system
CentOS 7.2 from `puppetlabs/centos-7.2-64-nocm` box

### Vagrantfile
```ruby
Vagrant.configure(2) do |config|
  config.vm.box = "puppetlabs/centos-7.2-64-nocm"
  config.vm.define "zabbixserver" do |zabbixserver|
    zabbixserver.vm.hostname = "zabbixserver.example.com"
    zabbixserver.vm.network :private_network, ip: "192.168.56.10"
    zabbixserver.vm.provider :virtualbox do |vm|
      vm.memory = 2048
      vm.cpus = 2
      vm.customize ["modifyvm", :id, "--ioapic", "on"]
    end
  end

  config.vm.define "zabbixproxy" do |zabbixproxy|
    zabbixproxy.vm.hostname = "zabbixproxy.example.com"
    zabbixproxy.vm.network :private_network, ip: "192.168.56.12"
    zabbixproxy.vm.provider :virtualbox do |vm|
      vm.memory = 2048
      vm.cpus = 2
      vm.customize ["modifyvm", :id, "--ioapic", "on"]
    end
  end

  config.vm.define "zabbixagent" do |zabbixagent|
    zabbixagent.vm.hostname = "zabbixagent.example.com"
    zabbixagent.vm.network :private_network, ip: "192.168.56.11"
    zabbixagent.vm.provider :virtualbox do |vm|
      vm.memory = 2048
      vm.cpus = 2
      vm.customize ["modifyvm", :id, "--ioapic", "on"]
    end
  end

  config.vm.define "zabbixdev" do |zabbixdev|
    zabbixdev.vm.hostname = "zabbixdev.example.com"
    zabbixdev.vm.network :private_network, ip: "192.168.56.13"
    zabbixdev.vm.provider :virtualbox do |vm|
      vm.memory = 2048
      vm.cpus = 2
      vm.customize ["modifyvm", :id, "--ioapic", "on"]
    end
  end
end
```

### Debug output
https://gist.github.com/vzabawski/1b87e1972a6d01aba04d6700b52cbfa1

### Expected behavior
vagrant-vbguest plugin is installed

### Actual behavior
Dependency error

### Steps to reproduce
1. Open shell
2. Run `vagrant plugin install vagrant-vbguest`

### References
Very similar problem without solution here:
- https://github.com/tmatilai/vagrant-proxyconf/issues/161

**Comment by robnov** [comment]
Also having this issue, but with VMWare plugin:

`Installing the 'vagrant-vmware-fusion' plugin. This can take a few minutes...
Bundler, the underlying system Vagrant uses to install plugins,
reported an error. The error is shown below. These errors are usually
caused by misconfigured plugin installations or transient network
issues. The error from Bundler is:

conflicting dependencies domain_name (= 0.5.20170404) and domain_name (= 0.5.20160826)
  Activated domain_name-0.5.20160826
  which does not match conflicting dependency (= 0.5.20170404)

  Conflicting dependency chains:
    domain_name (= 0.5.20160826), 0.5.20160826 activated

  versus:
    domain_name (= 0.5.20170404)

  Gems matching domain_name (= 0.5.20170404):
    domain_name-0.5.20170404`

**Comment by robnov** [comment]
Fixed by downloading the latest Vagrant (2.0.0) again, using the uninstall script to remove Vagrant (in MacOS, it's within the downloaded and unpacked vagrant .dmg). VMs remain untouched. Re-installing Vagrant 2.0.0, then all plugins. Works correctly now.

**Comment by chrisroberts** [comment]
@robnov Thanks for the update. There seems to be an issue of artifacts from previous version being left behind when upgrading which contributes to this issue. Will look at how we can ensure things are properly scrubbed on upgrade to prevent this issue.

**Comment by damienmckenna** [comment]
Ran into this problem too, uninstalling and reinstalling solved the problem for me.

**Comment by ZoranStojsavljevic** [comment]
This is working for me out of the box. Vagrant + VBox. Please, see my system setup, so you can make it stable! https://github.com/hashicorp/vagrant/issues/9396

For me, vagrant + libvirt combined do NOT work, since I have circular dependencies (the same link).

Zoran

**Comment by chrisroberts** [comment]
Installer fix for this has been shipped in the 2.0.3 release. Cheers!
